qcportal/qcportal/client_base.py

```diff
@@ -303,7 +305,24 @@ def _request(
             pretty_print_request(prep_req)
 
         try:
-            r = self._req_session.send(prep_req, verify=self._verify, timeout=self._timeout)
+            if allow_retries:
+                # Use the retry package. This will catch exceptions
+                r = retry_call(
+                    self._req_session.send,
+                    fkwargs={
+                        "request": prep_req,
+                        "verify": self._verify,
+                        "timeout": self._timeout,
+                    },
+                    exceptions=(requests.exceptions.ConnectionError, requests.exceptions.ConnectTimeout),
+                    tries=5,
+                    delay=0.5,
+                    max_delay=5,
+                    backoff=2,
```

Comment:
Instead of hardcoding these, perhaps use attributes on `PortalClientBase` configurable on `init`? Probably really only need to expose `tries`, `max_delay`, and `backoff`, but could expose all of these if you want to allow for customizability at runtime.

Also, probably need to set `jitter` as well to avoid many workers retrying all in-sync during a network hiccup.

Reply:
I'm fairly opposed to adding these to the constructor, since a vast, vast majority of people would use the defaults, and this would almost double the number of arguments available. I can do something similar to the `encoding` though and just allow those attributes to be set for a client object.